### ISSUE TITLE
[BUG] AzureBatch CloudTask.CopyNodeFileContent* methods are always failing with 'The specified pool does not exist.'

### ISSUE THREAD

**Issue by dterziev** [Azure/azure-sdk-for-net#6501]
**Describe the bug**
CloudTask.CopyNodeFileContent* methods are calling the wrong REST API and always fail.
A fix is available in pull request #6487.

***Exception or Stack Trace***
```text
Microsoft.Azure.Batch.Common.BatchException: Operation returned an invalid status code 'NotFound' ---> Microsoft.Azure.Batch.Protocol.Models.BatchErrorException: Operation returned an invalid status code 'NotFound'
   at Microsoft.Azure.Batch.Protocol.FileOperations.<GetFromComputeNodeWithHttpMessagesAsync>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.<ExecuteRequestWithCancellationAsync>d__42.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.<ExecuteRequestAsync>d__40.MoveNext()
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.<ExecuteRequestAsync>d__40.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.ProtocolLayer.<ProcessAndExecuteBatchRequest>d__85`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.ProtocolLayer.<GetNodeFileByNode>d__66.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.PoolOperations.<CopyNodeFileContentToStreamAsyncImpl>d__70.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.UtilitiesInternal.<ReadNodeFileAsStringAsync>d__33.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Batch.UtilitiesInternal.WaitAndUnaggregateException[T](Task`1 task)
   at Microsoft.Azure.Batch.UtilitiesInternal.WaitAndUnaggregateException[T](Task`1 task, IEnumerable`1 rootBehaviors, IEnumerable`1 additionalBehaviors)
   at Microsoft.Azure.Batch.CloudTask.CopyNodeFileContentToString(String filePath, Encoding encoding, GetFileRequestByteRange byteRange, IEnumerable`1 additionalBehaviors)
   at UserQuery.<Main>d__0.MoveNext() in C:\Users\terziev\AppData\Local\Temp\1\LINQPad5\_nzadqxho\query_efkhrs.cs:line 59
Request Information
ClientRequestId:
RequestId:114bc90d-7c2d-4961-937e-8b3516163ec9
HttpStatusCode:NotFound
StatusMessage:The specified pool does not exist.


Error Code = PoolNotFound, Lang=en-US, Message = The specified pool does not exist.
RequestId:114bc90d-7c2d-4961-937e-8b3516163ec9
Time:2019-06-06T10:28:00.5221962Z
Additional Values:
```

***Code Snippet***
```c#
var job = batchClient.JobOperations.GetJob("xxxx");
var tasks = job.ListTasks().ToList();
tasks[0].CopyNodeFileContentToString("stdout.txt");	//<-- this always throws
```

**Expected behavior**
Get the content of the file if the node is still around.

**Setup (please complete the following information):**
 - Microsoft.Azure.Batch 10.1.0

**Comment by kurtzeborn** [comment]
Thank you for opening this issue! We are routing it to the appropriate team for follow up.

**Comment by dterziev** [comment]
Just to clarify:
```c#
var job = batchClient.JobOperations.GetJob("xxxx");
var tasks = job.ListTasks().ToList();
tasks[0].CopyNodeFileContentToString("stdout.txt");	//<-- this always throws because internally it calls batchClient.PoolOperations.CopyNodeFileContentToStringAsyncImpl(...)

// Workaround, and arguably lower overhead approach, is to use:
batchClient.JobOperations.CopyNodeFileContentToString("xxxx", "1", "stdout.txt")
```

It boils down to the CloudTask class calling the [File - GetFromComputeNode API](https://docs.microsoft.com/en-us/rest/api/batchservice/file/getfromcomputenode) passing in the jobId in place of poolId, and taskId in place of computeNodeId:
```c#
return parentBatchClient.PoolOperations.CopyNodeFileContentToStringAsyncImpl(parentJobId, Id, filePath, encoding, byteRange, bhMgr, cancellationToken);

// and the signature for the PoolOperations method is:
class PoolOperations 
{
    internal Task<string> CopyNodeFileContentToStringAsyncImpl(string poolId, string computeNodeId, string filePath, Encoding encoding, GetFileRequestByteRange byteRange, BehaviorManager bhMgr, CancellationToken cancellationToken);
}
```

Instead it should be calling the [File - GetFileFromTask API](https://docs.microsoft.com/en-us/rest/api/batchservice/file/getfromtask) which is implemented in the JobOperations class

```c#
class JobOperations 
{
    internal Task<string> CopyNodeFileContentToStringAsyncImpl(string jobId, string taskId, string filePath, Encoding encoding, GetFileRequestByteRange byteRange, BehaviorManager bhMgr, CancellationToken cancellationToken);
}
```

The methods `CopyNodeFileContentToXXX` on `CloudTask` and `JobOperations` should have been named `CopyTaskFileContentToXXX`. Maybe good to consider renaming these in the next major version of the Microsoft.Azure.Batch Nuget package. The existing methods on CloudTask don't work anyway, so that isn't exactly a breaking change. Fixing the naming in `JobOperations` can be done by obsoleting the methods and replacing them with the correct named ones.

**Comment by matthchr** [comment]
@dterziev thanks for the report and the fix. This has been closed via #6487
